### PR TITLE
fix(core): Get rid of redundant logback dependency

### PR DESCRIPTION
**Pull request by flipp5b**
For now, `core` depends on logback. It's pretty uncommon for a library to depend on a concrete slf4j implementation (usually, it's up to the end application to decide which implementation to use).

This PR removes redundant logback dependency from the `core` and adds it to `examples`.

P.S. I'm new to Mill, so that may be not an idiomatic way to solve the issue :sweat_smile:

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -195,9 +194,14 @@ object akka extends Module {
 }
 
 object examples extends Module {
+  trait ExamplesJvmCommon extends Bot4sTelegramCrossPlatform {
+    override def ivyDeps = super.ivyDeps() ++ Agg(
```
Comment (flipp5b) [inline build.sc:198]: We could place here one more common part:
```scala
override val platformSegment: String = "jvm"
```

Comment (ex0ns) [reply build.sc:198]: Good catch, you can move that in the `ExamplesJvmCommon` trait indeed :ok_hand:

**Comment by ex0ns** [comment]
Thanks for the PR, I think that those chances are more than enough to solve this specific issue. 
By the way, you can test by running a few examples using the following command:
```scala
`mill -i __.examples.jvm[_].console`
new EchoBot("<Your telegram bot token>").run
```
And this works similarly for monix (replace `jvm` with `monixjvm`, `ziojvm` and `catsjvm`)

**Comment by flipp5b** [comment]
I've tested examples as follows:

```bash
mill examples.cats3jvm[_].run EchoBot <token>
```

I can't spot any issues, at least logging works.

**Comment by ex0ns** [comment]
@flipp5b this is available in release 5.8.1 that should soon be available

**Pull request merged**